### PR TITLE
LDAP Configuration

### LINKED ISSUE #213: Recipes: Removal
### Cookbook version
4.0.0-beta

### Chef-client version
Latest

### Platform Details
- All supported

### Scenario:
As a user
I do not want to be using any recipes or attributes, all required properties should be using custom resources

### Expected Result:
- [ ] grafana is configurable via resource properties
- [ ] grafana has unit tests 
- [ ] there are no more recipes in grafana
- [ ] there are no more attributes in grafana

### PR DESCRIPTION
**Pull request by xorima**
### Description

Enables us to configure ldap using resources and removes the need to attributes and recipes from the cookbook

### Issues Resolved

#213 

### Check List
- [x] All tests pass. See https://github.com/sous-chefs/grafana/blob/master/TESTING.md
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: resources/config_ldap_group_mappings.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#
+# Cookbook Name:: grafana
+# Resource:: config_alerting
+#
+# Copyright 2018, Sous Chefs
```
Comment (damacus) [inline resources/config_ldap_group_mappings.rb]: 2019

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by xorima** [comment]
Documentation will be done as a specific pr

File: resources/config_ldap.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# limitations under the License.
+#
+# Configures the installed grafana instance's ldap settings
+# See https://raw.githubusercontent.com/grafana/grafana/master/conf/ldap.toml
+
+property  :instance_name,                 String, name_property: true
+property  :conf_directory,                String, default: '/etc/grafana'
+property  :config_file,                   String, default: lazy { ::File.join(conf_directory, 'ldap.toml') }
+property  :log_filters,                   String
+property  :servers_attributes_name,       String, default: 'givenName'
```
Comment (JohnRoesler) [inline resources/config_ldap.rb:26]: Could you make `server_attributes` a hash for name, surname, etc. as keys?

Comment (xorima) [reply resources/config_ldap.rb:26]: Could do but so far everything has been an explicit property, feels like it would be an oddball in this cookbook?

Comment (JohnRoesler) [reply resources/config_ldap.rb:26]: Yeah, consistent feel seems better

**Review by JohnRoesler — APPROVED** [review]

**Pull request merged**